plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/Storage.java

```diff
@@ -65,8 +65,10 @@ public Optional<String> getOptionalLocation()
 
     public String getLocation()
     {
-        // Default getter requires location to be set. Location is not set only in rare case when in CREATE TABLE flow in Hive connector when
-        // delegate-transactional-managed-table-location-to-metastore is set to true and location is determined by HMS.
+        // Default getter requires location to be set. Location is not set in the following scenarios:
+        // 1. CREATE TABLE flow in Hive connector sets delegate-transactional-managed-table-location-to-metastore to true and location is determined by HMS.
+        // 2. Iceberg format tables
+        // 3. Delta format tables (sometimes)
```

Comment:
I double checked this today. Looks like we always expect the path to be in the storage descriptor. We could change that for Managed tables, but right now it's required.

Reply:
@alexjo2144, do you want me to remove the assumption that storage won't be specified for delta tables then?

Reply:
I think this change is fine, I made an Issue for follow up https://github.com/trinodb/trino/issues/12180